docs/source/notebooks/Boson_Bunching.ipynb

```diff
@@ -0,0 +1,1076 @@
+    "\n",
+    "\n",
+    "\n",
+    "![Blochsphere](../_static/img/Blochsphere.png)\n",
+    "\n",
+    "Here one can see the two fully distinguishable photons with red arrows and the five linear combinations in the equator plane."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def create_inputs(n,w):\n",
+    "    #make a list of fully distinguishable photons of the form |{a:i}> that means we give the photon an attribute a with value i \n",
+    "    states_dist = []\n",
+    "    for i in range(1,n+1):\n",
+    "        x = \"|{{a:{}}}>\".format(i)\n",
+    "        states_dist.append(pcvl.StateVector(x))\n",
+    "    #make a list of all input photons which are superpositions of |{a:1}> and |{a:2}>\n",
+    "    states_par_dist = [states_dist[0], states_dist[1]] #start with the pure states |{a:1}> and |{a:2}>\n",
+    "    for i in range(2,n):\n",
+    "        x = states_par_dist[0] +  states_par_dist[1] * w**(i-2) #add the states |{a:1}> + w^(i-2)|{a:2}>\n",
+    "        states_par_dist.append(x)\n",
+    "    #Input states\n",
+    "    #initialise the variables\n",
+    "    indisting_photons = [] \n",
+    "    par_disting_photons = 1 \n",
+    "    disting_photons = 1 \n",
+    "    #fill the states\n",
+    "    for i in range(n):\n",
+    "        indisting_photons.append(1) # gives the state |1,1, ... , 1>\n",
+    "        par_disting_photons = par_disting_photons * states_par_dist[i]\n",
+    "        disting_photons = disting_photons * states_dist[i] # gives the state |{a:1},{a:2}, ... ,{a:n}> \n",
+    "    return pcvl.BasicState(indisting_photons), par_disting_photons, disting_photons\n",
+    "\n",
+    "\n",
+    "input_indistinguishable, input_partiallydistinguishable, input_distinguishable = create_inputs(n,w) #indistinguishable photons, partially distinguishable photons, fully distinguishable photons respectively "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To see if photon bunching gets amplified for partially distiguishable photons we need to compare it to the case of indistinguishable photons and for interest we are also comparing it to fully distinguishable photons. We are expecting that the probability for partially distinguishable photons will be highest followed by the indistiguishable photons and fully distinguishable photons having the lowest probability."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#simulating boson sampling\n",
+    "p = pcvl.Processor(\"SLOS\")\n",
+    "p.set_circuit(circuit_rand)\n",
+    "s = pcvl.SimulatorFactory().build(p)"
```

Comment:
It's usually better to use complete word for variable (unless it's a mathematical stuff like n, m, U ...)
So maybe use `processor` or `proc` instead of `p` and `simulator` or `sim` instead of `s`